common/src/main/java/com/wynntils/wynn/model/container/ContainerQueryManager.java

```diff
@@ -58,24 +62,40 @@ public static void runQuery(ContainerQueryStep firstStep) {
 
         currentStep = firstStep;
         firstStepName = firstStep.getName();
-
+        lastStep = null;
+        resetTimer();
         if (!firstStep.startStep(null)) {
             raiseError("Cannot execute first step");
         }
     }
 
     @SubscribeEvent
     public static void onMenuOpened(MenuEvent.MenuOpenedEvent e) {
-        if (currentStep == null) return;
+        if (currentStep == null && lastStep == null) return;
+        if (currentStep == null && lastStep != null) {
+            // We're in a possibly bad state. We have failed a previous call, but
+            // we might still get the menu opened (perhaps after a lag spike).
+            boolean matches = lastStep.verifyContainer(e.getTitle(), e.getMenuType());
+            if (matches) {
+                WynntilsMod.warn("Closing container '" + e.getTitle().getString()
```

Comment:
The container isn't actually closed?

Reply:
We do ` e.setCanceled(true)` on the open request. This means the client will not see this as a request to open a container. So technically it's more "prohibit that it does not open" than "close", but that did not make a catchy error message :)

Reply:
Yes, but this is fired in response to a packet from the server. If we cancel the event, that just means we don't handle the packet client side, while the server still believes that the container is open.